### DISCUSSION TITLE
Importing from Gitlab API

### DISCUSSION THREAD

**Discussion by ghost** [microcks/microcks#973] [Q&A] [answered]
How do we import specs from a gitlab repository?

**Comment by lbroudoux** [comment, marked as answer]
Importing from GItLab requires using the API URL instead of the RAW URL. You'll need the GitLab project ID for your repo and your instance API URL to do that.

On a public GitLab instance, you may end with something like: `https://gitlab.com/api/v4/projects/<project_id>/repository/files/api-contracts%2FUserRegistrationAPI-openapi-1.0.0.yml/raw?ref=main``

where the path to your OpenAPI file is URL encoded (see https://www.urlencoder.io/  if you need help) and the branch is specified as the `ref` query parameter.

Hope it helps!
